src/main/java/com/epam/ta/reportportal/core/launch/impl/FinishLaunchHandlerAsyncImpl.java

```diff
@@ -0,0 +1,95 @@
+package com.epam.ta.reportportal.core.launch.impl;
+
+import com.epam.ta.reportportal.commons.ReportPortalUser;
+import com.epam.ta.reportportal.core.launch.FinishLaunchHandler;
+import com.epam.ta.reportportal.core.launch.util.LaunchLinkGenerator;
+import com.epam.ta.reportportal.entity.launch.Launch;
+import com.epam.ta.reportportal.ws.model.BulkRQ;
+import com.epam.ta.reportportal.ws.model.FinishExecutionRQ;
+import com.epam.ta.reportportal.ws.model.OperationCompletionRS;
+import com.epam.ta.reportportal.ws.model.launch.FinishLaunchRS;
+import com.epam.ta.reportportal.ws.rabbit.MessageHeaders;
+import org.springframework.amqp.core.AmqpTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.epam.ta.reportportal.core.configs.RabbitMqConfiguration.QUEUE_LAUNCH_BULK_STOP;
+import static com.epam.ta.reportportal.core.configs.RabbitMqConfiguration.QUEUE_LAUNCH_FINISH;
+import static com.epam.ta.reportportal.core.configs.RabbitMqConfiguration.QUEUE_LAUNCH_STOP;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * @author Konstantin Antipin
+ */
+@Service
+@Qualifier("finishLaunchHandlerAsync")
+public class FinishLaunchHandlerAsyncImpl implements FinishLaunchHandler {
+
+    @Autowired
+    @Qualifier(value = "rabbitTemplate")
+    AmqpTemplate amqpTemplate;
+
+    @Override
+    public Launch finishLaunch(String launchId, FinishExecutionRQ request, ReportPortalUser.ProjectDetails projectDetails,
+                               ReportPortalUser user) {
+        throw new UnsupportedOperationException("Async api unsupported operation");
```

Comment:
?

Reply:
Looks like this method out of two finishLaunch() in FinishLaunchHandler is not appropriate for async implementation, because it returns Launch, and we don't get entity from DB in async implementation. So it wouldn't be ever called. Instead the second finishLaunch() constructs and provides response.
Any idea how to do it otherwise?